include/revng/ADT/STLExtras.h

```diff
@@ -211,9 +211,14 @@ skipImpl(IteratorType &&From,
 
   std::ptrdiff_t TotalSkippedCount = Front + Back;
   if constexpr (forward_iterator<IteratorType>) {
-    // We cannot compute the assert on the input iterators because it's
```

Comment:
Is this correct? Don't you need a random access iterator (or at very least a bidirectional iterator) to compute `std::distance` without irreversibly consume the iterators?

Reply:
Only `input` and `ouput` iterators are "one-time" deal. All the others are safe as long as you do it on a copy of the iterator which we are doing here.

Reply:
Ahhh, right!